crates/ide-completion/src/context.rs

```diff
@@ -569,6 +569,28 @@ impl<'a> CompletionContext<'a> {
         // completing on
         let original_token = original_file.syntax().token_at_offset(offset).left_biased()?;
 
+        // try to skip completions on path with qinvalid colons
+        // this approach works in normal path and inside token tree
+        match original_token.kind() {
+            T![:] => {
+                // return if no prev token before colon
+                let prev_token = original_token.prev_token()?;
+
+                // only has a single colon
+                if prev_token.kind() != T![:] {
+                    return None;
+                }
+
+                if !is_prev_token_valid_path_start_or_segment(&prev_token) {
+                    return None;
+                }
+            }
+            T![::] if !is_prev_token_valid_path_start_or_segment(&original_token) => {
```

Comment:
I don't think we need to check for the `is_prev_token_valid_path_start_or_segment` stuff, `::` can follow a lot of different characters just fine, `}::` for example works, in whic hcase it just starts an absolute path.

Reply:
This check is similar for the case in https://github.com/rust-lang/rust-analyzer/pull/13611#discussion_r1027110327 but happens less frequently. As you pointed out current check is insufficient so it's seems better to just remove this one.